crates/client/gateway/src/client/request_builder.rs

```diff
@@ -0,0 +1,109 @@
+use std::collections::HashMap;
+
+use mp_block::{BlockId, BlockTag};
+use reqwest::Client;
+use serde::de::DeserializeOwned;
+use starknet_types_core::felt::Felt;
+use url::Url;
+
+use crate::error::{SequencerError, StarknetError};
+
+#[derive(Debug, Clone)]
+pub struct RequestBuilder<'a> {
+    client: &'a Client,
+    url: Url,
+    params: HashMap<String, String>,
+    headers: HashMap<String, String>,
```

Comment:
nit(perf): i'd love it if you used HeaderName and HeaderValue from tower http, cause they almost never allocate on the happy path and that's better

also their HeaderMap

Reply:
idk if it makes sense with reqwest though

Reply:
for params, if possible tower's http uri builder to avoir allocs too but it's nitpick too

i don't like how we are doing sooo many allocations here for a simple builder, it's a bit dumb -- but starknet providers also did that

Reply:
i think for now just use [`http`](https://docs.rs/http/latest/http/)'s HeaderMap and for params, a HashMap<Cow<'static, str>, Cow<'static, str>> with `add_param(mut self, name: impl Into<Cow<'static, str>>, value: impl Into<Cow<'static, str>>)` so that adding a param doesnt allocate when the string is static